Sources/CNIOLinux/include/liburing_stubs.h

```diff
@@ -0,0 +1,201 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LIBURING_STUBS_H
+#define LIBURING_STUBS_H
+
+// only read below stubs if liburing headers were missing
+// stubs are needed to build on systems lacking liburing
+
+#ifdef C_NIO_LIBURING_UNAVAILABLE
+
+#ifdef __linux__
+
+#include <stdbool.h>  // bool
+
+// #include <linux/time_types.h> // struct __kernel_timespec - only for newer distros, copied def below for compile
```

Comment:
we'll need to remove the copies of the structs here, they'll need to match the runtime of the target system.

Reply:
Same as above - those are never used in reality (if that include is pulled in, it means that there is a macro defined that also disables uring permanently from the framework). struct __kernel_timespec is only available in newer kernel / distributions so can't pull it in if we want to compile on older versions.

Reply:
Right, I think this will be a problem though if we don't have uring but we do have `struct __kernel_timespec`. Then we do a redefinition of it which is an error.

Reply:
Ok, will add a can include macro check then and pull it in from linux/time_types.h if it exists, otherwise define - then it should be robust for both.